api/src/main/java/org/openmrs/module/reporting/config/ReportLoader.java

```diff
@@ -0,0 +1,251 @@
+    }
+
+    public static void saveReportDesigns(ReportDefinition reportDefinition, List<ReportDesign> reportDesigns) {
+        // purging a ReportDesign doesn't trigger any extra logic, so we can just purge-and-recreate here
+        List<ReportDesign> existingDesigns = Context.getService(ReportService.class).getReportDesigns(reportDefinition, null, true);
+        if (existingDesigns.size() > 0) {
+            log.debug("Deleting " + existingDesigns.size() + " old designs for " + reportDefinition.getName());
+            for (ReportDesign design : existingDesigns) {
+                Context.getService(ReportService.class).purgeReportDesign(design);
+            }
+        }
+
+        for (ReportDesign reportDesign : reportDesigns) {
+            Context.getService(ReportService.class).saveReportDesign(reportDesign);
+        }
+
+    }
+
+    public static  ReportDefinition constructReportDefinition(ReportDescriptor reportDescriptor) {
+
+        ReportDefinition rd = new ReportDefinition();
+        // TODO do we need to handle translations here?
+        rd.setName(reportDescriptor.getName());
+        rd.setDescription(reportDescriptor.getDescription());
+        rd.setUuid(reportDescriptor.getUuid());
+
+        List<Parameter> parameters = constructParameters(reportDescriptor.getParameters());
+        Map<String, Object> mappings = constructMappings(parameters);
+        rd.setParameters(parameters);
+
+        if (reportDescriptor.getDatasets() != null) {
+            for (DataSetDescriptor dataSetDescriptor : reportDescriptor.getDatasets()) {
+                DataSetDefinition dsd = constructDataSetDefinition(dataSetDescriptor, parameters);
+                // TODO do we need to handle DSD name and description?
+                if (dsd != null) {
+                    rd.addDataSetDefinition(dataSetDescriptor.getKey(), dsd, mappings);
+                }
+            }
+        }
+        return rd;
+    }
+
+    public static List<Parameter> constructParameters(List<ParameterDescriptor> parameterDescriptors) {
+        List<Parameter> parameters = new ArrayList<Parameter>();
+        if (parameterDescriptors != null) {
+            for (ParameterDescriptor parameterDescriptor : parameterDescriptors) {
+                try {
+                    // TODO how else might we want to configure parameters? ie other "types" besides classes?
+                    parameters.add(new Parameter(parameterDescriptor.getKey(), parameterDescriptor.getLabel(), Class.forName(parameterDescriptor.getType())));
+                } catch (Exception e) {
+                    log.error("Unable to configure parameter " + parameterDescriptor.getKey(), e);
+                }
+            }
+        }
+        return parameters;
+    }
+
+
+    public static Map<String, Object> constructMappings(List<Parameter> parameters) {
+        // TODO: Mike is the correct and/or necessary?
```

Comment:
@mseaton  just a ping, can you confirm this is correct?

Reply:
yes, this is both correct and necessary